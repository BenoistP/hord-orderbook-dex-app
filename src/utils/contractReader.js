import { store } from 'index'
import { createNotification } from 'store/actions/uiActions'

export const useContractReader = async (
  contract,
  contractFunction,
  parameters = [],
  callMethod = (contractInstance) => contractInstance?.[contractFunction](...parameters)?.call(),
) => {
  let result
  try {
    result = await callMethod(contract)
  } catch (error) {
    result = { error: error.message }
    store.dispatch(createNotification('error', 'Something went wrong reading data from the blockchain.', 4000))
  }
  return result
}
