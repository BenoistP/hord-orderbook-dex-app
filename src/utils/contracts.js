import MatchingMarket from './contracts/MatchingMarket.json';
import UniswapSimplePriceOracle from './contracts/UniswapSimplePriceOracle.json';
import ERC20 from './contracts/ERC20.json';
import MakerOtcSupportMethods from './contracts/MakerOtcSupportMethods.json';

export const contracts = [
  {
    name: 'MatchingMarket',
    address: MatchingMarket.address,
    abi: MatchingMarket.abi,
    link: `https://testnet.bscscan.com/address/${MatchingMarket.address}`,
  },
  {
    name: 'UniswapSimplePriceOracle',
    address: UniswapSimplePriceOracle.address,
    abi: UniswapSimplePriceOracle.abi,
    link: `https://testnet.bscscan.com/address/${UniswapSimplePriceOracle.address}`,
  },
  {
    name: 'MakerOtcSupportMethods',
    address: MakerOtcSupportMethods.address,
    abi: MakerOtcSupportMethods.abi,
    link: `https://testnet.bscscan.com/address/${MakerOtcSupportMethods.address}`,
  },
  {
    name: 'BUSD',
    address: '0x78867bbeef44f2326bf8ddd1941a4439382ef2a7',
    abi: ERC20.abi,
    link: `https://testnet.bscscan.com/address/0x78867bbeef44f2326bf8ddd1941a4439382ef2a7`,
  },
  // {
  //   name: 'HPoolToken',
  //   address: '0x39a8f0f941bcb56ff0f19371a366496dfc948704',
  //   abi: ERC20.abi,
  //   link: `https://testnet.bscscan.com/address/0x39a8f0f941bcb56ff0f19371a366496dfc948704`,
  // },
];
