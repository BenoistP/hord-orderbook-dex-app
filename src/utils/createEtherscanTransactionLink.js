import { KOVAN } from "../constants";

const createEtherscanTransactionLink = ({
  activeNetworkName,
  transactionHash
}) => {
  switch (activeNetworkName) {
    case KOVAN:
      return `https://${activeNetworkName}.etherscan.io/tx/${transactionHash}`;
    default:
      return `https://etherscan.io/tx/${transactionHash}`;
  }
};

export default createEtherscanTransactionLink;
