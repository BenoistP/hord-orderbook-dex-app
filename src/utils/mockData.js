export const hpoolList = [{
    name: 'hpool1',
    address: '0x39a8f0f941bcb56ff0f19371a366496dfc948704',
    image: "https://hord-prod-users.s3.amazonaws.com/media/profile/93ca4033-5c01-4790-8ce1-a78943e0feb3.jpg"
},
{
    name: 'hpool2',
    address: '0x39a8f0f941bcb56ff0f19371a366496dfc948704', // duplicate => TODO: add more tokens
    image: "https://hord-prod-users.s3.amazonaws.com/media/profile/93ca4033-5c01-4790-8ce1-a78943e0feb3.jpg"
}]