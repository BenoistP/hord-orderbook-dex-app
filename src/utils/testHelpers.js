import MockDate from 'mockdate';
import moment from 'moment-timezone';

export function mockDate(date, block) {
  try {
    MockDate.set(date);
    moment.tz.setDefault('UTC');
    return block.call();
  } finally {
    MockDate.reset();
    moment.tz.setDefault(moment.tz.guess());
  }
}
