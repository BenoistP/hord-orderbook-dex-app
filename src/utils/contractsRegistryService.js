import { contracts } from './contracts';

const createContract = (address, abi) => async () => {
  return new window._web3.eth.Contract(abi, address);
};

export const loadContracts = async () => {
  const loadedContracts = {};
  await Promise.all(contracts.map(async (contract) => {
    loadedContracts[contract.name] = await createContract(contract.address, contract.abi)();
    return contract;
  }));

  return loadedContracts;
}