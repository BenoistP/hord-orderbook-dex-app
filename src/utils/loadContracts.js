import { Contract } from 'ethers';
import { contracts } from './contracts';

export const loadContract = (address, abi, signer) => {
  return new Contract(
    address,
    abi,
    signer,
  );
};

export const loadContracts = (signer) => {
  const loadedContracts = {};
  contracts.map((contract) => {
    loadedContracts[contract.name] = loadContract(contract.address, contract.abi, signer);
    return contract;
  });

  return loadedContracts;
}

