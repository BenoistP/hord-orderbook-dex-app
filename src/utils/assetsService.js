import Dec from 'decimal.js';
import { getErc20Contract as _ } from './contractsRegistryService';

const assetAmountInEth = (amount) => {
  const decimals = 18;
  return Dec(amount?.toString() || 0)
    .div(10 ** decimals)
    .toString();
};

const getAssetAllowanceForAddress = async (owner, spender, contract) => {
  debugger
  const data = await contract.allowance(owner, spender).call();
  debugger

  return assetAmountInEth(data);
};

export const isAddressApprovedOnAsset = async (account, addressToCheck, contract, amount) => {
  const allowance = await getAssetAllowanceForAddress(account, addressToCheck, contract);
  return Dec(allowance).gte(amount);
};