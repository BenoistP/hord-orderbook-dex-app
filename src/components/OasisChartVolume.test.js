/* global describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { mockDate } from '../utils/testHelpers'

import {
  OasisChartVolume,
  mapStateToProps,
  mapDispatchToProps,
} from './OasisChartVolume';

describe('(Component) OasisChartVolume', () => {
  const state = fromJS(global.storeMock);
  console.log('original', new Date(), new Date().getTimezoneOffset())
  const initialProps = mockDate('2018-05-10', () => {
    console.log('mock', new Date(), new Date().getTimezoneOffset())
    return mapStateToProps(state)
  });
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps,
    tradingPair: {
      baseToken: 'empty1',
      quoteToken: 'empty2',
    },
  };

  it('should render', () => {
    const wrapper = shallow(
      <OasisChartVolume {...props}/>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
