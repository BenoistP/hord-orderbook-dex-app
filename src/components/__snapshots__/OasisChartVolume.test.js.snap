// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OasisChartVolume should render 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(140, 133, 200, 0.1)",
          "borderColor": "#8D86C9",
          "borderWidth": 3,
          "data": Array [
            "0.00",
            "0.00",
            "0.00",
            "0.00",
            "0.00",
            "0.00",
            "0.00",
          ],
          "label": "Volume",
          "pointBackgroundColor": "#8D86C9",
          "pointRadius": 3,
        },
      ],
      "labels": Array [
        1525384800,
        1525471200,
        1525557600,
        1525644000,
        1525730400,
        1525816800,
        1525903200,
      ],
    }
  }
  options={
    Object {
      "layout": Object {
        "padding": 5,
      },
      "legend": Object {
        "display": false,
      },
      "scales": Object {
        "xAxes": Array [
          Object {
            "display": false,
          },
        ],
        "yAxes": Array [
          Object {
            "ticks": Object {
              "beginAtZero": true,
            },
          },
        ],
      },
      "tooltips": Object {
        "custom": [Function],
        "enabled": false,
        "mode": "index",
        "position": "nearest",
      },
    }
  }
/>
`;
